File: benchmarks/fixtures/webapp_swift/Sources/webapp/routes/AuthRoutes.swift
import Foundation

/// Wires auth endpoints to their handlers.
final class AuthRoutes {
    private let handlers: AuthHandlers
    private let middleware: AuthMiddleware

    init(handlers: AuthHandlers, middleware: AuthMiddleware) {
        self.handlers = handlers
        self.middleware = middleware
    }

    func register() {
        getLogger("AuthRoutes").info("registered /login, /logout")
    }

    func dispatchLogin(email: String, password: String) -> String {
        return handlers.handleLogin(email: email, password: password)
    }
}

File: benchmarks/fixtures/webapp_kt/src/main/kotlin/webapp/routes/AuthRoutes.kt
package webapp.routes

import webapp.api.AuthHandlers
import webapp.middleware.AuthMiddleware
import webapp.util.getLogger

/** Wires auth endpoints to their handlers. */
class AuthRoutes(
    private val handlers: AuthHandlers,
    private val middleware: AuthMiddleware
) {
    fun register() {
        getLogger("AuthRoutes").info("registered /login, /logout")
    }

    fun dispatchLogin(email: String, password: String): String {
        return handlers.handleLogin(email, password)
    }
}
